.github/workflows/ci.yml

```diff
@@ -0,0 +1,445 @@
+    container:
+      image: ghcr.io/matrixai/github-runner
+
+    needs:
+      - build-dist
+
+    if: github.ref == 'refs/heads/staging'
+
+    steps:
+      - uses: actions/checkout@v4
+      - run: entrypoint
+
+      - name: Build dist
+        run: |
+          echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ./.npmrc
+          echo 'Publishing library prerelease'
+
+          nix develop .#ci --command bash -c $'
+            npm publish --tag prerelease --access public
+          '
+
+          rm -f ./.npmrc
+
+  integration-builds:
+    name: "Integration / Builds"
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/matrixai/github-runner
+
+    needs:
+      - build-pull
+      - build-dist
+
+    if: github.ref == 'refs/heads/staging'
+
+    steps:
+      - uses: actions/checkout@v4
+      - run: entrypoint
+
+      - name: Build targets
+        run: |
+          mkdir -p ./builds
+
+          build-application="$(nix build \
+            --max-jobs "$(nproc)" \
+            --cores "$(nproc)" \
+            --print-out-paths \
+            .# \
+          )"
+
+          nix-store --export $( \
+            nix-store --query --requisites "$build-application" \
+          ) | gzip > ./builds/polykey-cli.closure.gz
+
+          builds="$(nix build \
+            --max-jobs "$(nproc)" \
+            --cores "$(nproc)" \
+            --print-out-paths \
+            --print-build-logs \
+            .# \
+            .#docker \
+            .#packages.x86_64-linux.executable \
+          )"
+
+          cp -r $(echo $builds | tr '\n' ' ') ./builds/
+
+          nix develop .#ci --command bash -c $'
+            npm run build
+          '
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: builds
+          path: ./builds
+
+  integration-deployment:
+    name: "Integration / Deployment"
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/matrixai/github-runner
+
+    concurrency:
+      group: integration-deployment
+      cancel-in-progress: true
+
+    needs: integration-builds
+
+    if: github.ref == 'refs/heads/staging'
+
+    steps:
+      - name: Deploying image to ECR
+        run: |
+          nix develop .#ci --command bash -c $'
+            aws ecr get-login-password \
+              | skopeo login \
+              --username AWS \
+              --password-stdin \
+              --authfile "$REGISTRY_AUTH_FILE" \
+              "$CI_REGISTRY_IMAGE"
+
+            image=(./builds/*-docker-*)
+
+            ./scripts/deploy-image.sh "${image[0]}" \'testnet\' "$CI_REGISTRY_IMAGE"
+          '
+
+          echo 'Waiting for Testnet Deployment'
+
+          nix develop .#ci --command bash -c $'
+            ./scripts/wait-for-deploy.js testnet.polykey.com
+          '
+
+  integration-platforms:
+    name: "Integration / Platforms"
+    runs-on: ubuntu-latest
+
+    concurrency:
+      group: integration-platforms
+      cancel-in-progress: true
```

Comment:
Same here. Why should in-progress testing on platforms be cancelled if a new one is triggered?